instant-xml-macros/src/de.rs

```diff
@@ -34,6 +34,25 @@ impl quote::ToTokens for Deserializer {
 impl Deserializer {
     pub fn new(input: &syn::DeriveInput) -> Deserializer {
         let ident = &input.ident;
+        let generics = (&input.generics).into_token_stream();
```

Comment:
Probably want to use `split_for_impl()` here, too. I recently wrote something similar like this:

```rust
    let mut email_generics = ast.generics.clone();
    let email_lifetime = syn::LifetimeDef::new(syn::Lifetime::new("'email", Span::call_site()));
    for lt in email_generics.lifetimes_mut() {
        lt.bounds.push(email_lifetime.lifetime.clone());
    }
    email_generics
        .params
        .push(syn::GenericParam::Lifetime(email_lifetime));

    // Split the generics for use in impls and type definitions, below.

    let (_, inner_ty_generics, _) = ast.generics.split_for_impl();
    let (impl_generics, ty_generics, where_clause) = email_generics.split_for_impl();
```

(You may not need both of the split sets.)

Reply:
Unfortunately, it is not very helpful here, at the end we would need something similar to what serde is doing [here](https://github.com/serde-rs/serde/blob/f52d134c14f6904010d0a59107987b050b36b811/serde_derive/src/bound.rs). But it will take more time to implement, so I would leave it as it is for now.

Reply:
Why is it not helpful here?

Reply:
Because what I can get useful from `split_for_impl` is basically this:` let generics = (&input.generics).into_token_stream();`
What I need is: 
e.g.
": 'a + 'b, 'a, 'b" and ", 'a, 'b"

This is not about getting the generics from the struct is about combining them with our 'xml lifetime to have something like:
` struct StructVisitor<'xml: 'a + 'b, 'a, 'b> {
            marker: std::marker::PhantomData<StructDeserializerScalars<'a, 'b>>,
            lifetime: std::marker::PhantomData<&'xml ()>,
        }
        impl<'xml: 'a + 'b, 'a, 'b> Visitor<'xml> for StructVisitor<'xml, 'a, 'b> {`
        Unless you are thinking about something I don't get?

Reply:
Still, don't get how to get something like`<'xml: 'a + 'b, 'a, 'b>` from it. I will try to look at it once more later.

Reply:
I definitely think we need `split_for_impl()` here, but before we do that we probably need to modify the original `Generics` to add the `'xml` lifetime.

Reply:
Will take care of this separately.